blackbird_python/blackbird/listener.py

```diff
@@ -561,7 +561,8 @@ def exitProgram(self, ctx: blackbirdParser.ProgramContext):
         self._program._var.update(_VAR)
         _VAR.clear()
 
-        self._program._parameters.extend(_PARAMS)
+        is_ptype = lambda p: str(p)[0] == "p" and str(p)[1:].isdigit()
```

Comment:
Very minor suggestion, but you could define this function at the module level, since it doesn't seem to depend on closure :)